### PR TITLE
Refactor hardcoded color to use OUI in `region_map`

### PR DESCRIPTION
**Pull request by BSFishy**
### Description

Changes a hardcoded color to use a color from the OUI theme. This color seems visually similar to `ouiColorMediumShade`, so I changed it to that. Need guidance from UX on whether this is the correct choice @KrooshalUX 

### Issues Resolved

Partial js component of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4289

## Screenshot

**Original**
Screenshot (78)

**New**
Screenshot (77)

**Set to pink for reference**
Screenshot (74)

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by manasvinibs — COMMENTED** [review]
LGTM! Waiting for UX review to approve.

File: src/plugins/region_map/public/choropleth_layer.js
Comment (joshuarrrr) [inline src/plugins/region_map/public/choropleth_layer.js:50]: This is kind of an interesting case. From the sample screenshots, I'd also make the leap to guess that it's used for both the outline and fill color (but would need a deeper dive into the code to verify), and is intended to be a neutral default. However, given it's usage, I _think_ something like `ouiPaletteColorBlind()[1]` (or a seed color from one of the quantitative palettes, like `ouiPaletteGray()`) may be more appropriate for cohesion with other visualizations. 

To help us decide, @BSFishy can you provide more info about how/when this `EMPTY_STYLE` is used? And confirm whether the `color` prop is only used for strokes or also for fill?

**Comment by KrooshalUX** [comment]
I do think its a _little dark_ making the borders feel very heavy - and I am interested in knowing potential impacts from @joshuarrrr  question before I approved.

File: src/plugins/region_map/public/choropleth_layer.js
Comment (BSFishy) [reply src/plugins/region_map/public/choropleth_layer.js:50]: This style is simply passed to leaflet, and I can now confirm it is only used as the stroke color: https://leafletjs.com/reference.html#path-color

**Review by AMoo-Miki — APPROVED** [review]

File: src/plugins/region_map/public/choropleth_layer.js
Comment (joshuarrrr) [reply src/plugins/region_map/public/choropleth_layer.js:50]: Thanks for the context, that's helpful.

**Comment by joshuarrrr** [comment]
All checks pass.

@KrooshalUX See https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4299#discussion_r1240434831 for updated usage context.

**Comment by KrooshalUX** [comment]
While the ouiColorMediumShade is still too dark, IMO, we run greater risk choosing any other grey as the change between light and dark mode is more drastic. 

Lets ship this for now, I think we need to introduce some vis specific border sass color variables in the future that are not tied to forms and stay relatively neutral regardless of light/dark mode theme selection.

**Comment by joshuarrrr** [comment]
Cypress task actually passed:

```
  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  core-opensearch-dashboards/opensear      01:16        5        5        -        -        - │
  │    ch-dashboards/apps/vis_builder/basi                                                         │
  │    c.spec.js                                                                                   │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:43        4        4        -        -        - │
  │    ch-dashboards/apps/vis_builder/dash                                                         │
  │    board.spec.js                                                                               │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:43        4        4        -        -        - │
  │    ch-dashboards/apps/vis_builder/expe                                                         │
  │    rimental.spec.js                                                                            │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:28        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/area.spec.js                                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:27        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/bar.spec.js                                                                           │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:27        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/line.spec.js                                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:25        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/metric.spec.js                                                                        │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:30        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/table.spec.js                                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:32        4        4        -        -        - │
  │    ch-dashboards/apps/vis_type_table/b                                                         │
  │    asic.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      01:29        9        9        -        -        - │
  │    ch-dashboards/apps/vis_type_table/d                                                         │
  │    ata.spec.js                                                                                 │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      01:22        4        4        -        -        - │
  │    ch-dashboards/apps/vis_type_table/e                                                         │
  │    mbed.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      01:01       11       11        -        -        - │
  │    ch-dashboards/apps/vis_type_table/o                                                         │
  │    ptions.spec.js                                                                              │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear       28ms        7        -        -        7        - │
  │    ch-dashboards/apps/vis_type_table/s                                                         │
  │    plit.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      01:23       34       34        -        -        - │
  │    ch-dashboards/dashboard_sanity_test                                                         │
  │    _spec.js                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        2ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/1_create_datasource.spec.js                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        2ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/2_datasource_table.spec.js                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        1ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/3_update_datasource.spec.js                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        10:53       87       80        -        7        -  

Done in 857.82s.
```

**Review by joshuarrrr — DISMISSED** [review]
BTW, one reason we can't effectively use one of the theme-based grayscale values is that they're inverted depending on whether you're in dark or light mode (so that ` ouiColorDarkestShade` in dark mode is actually the _lightest_ shade of gray 🤯 ). And it doesn't make sense to use theme driven colors until/unless we use dark-mode map tiles, which provide the relevant background in this case.

**Review by manasvinibs — DISMISSED** [review]

**Comment by KrooshalUX** [comment]
> BTW, one reason we can't effectively use one of the theme-based grayscale values is that they're inverted depending on whether you're in dark or light mode (so that ` ouiColorDarkestShade` in dark mode is actually the _lightest_ shade of gray 🤯 ). And it doesn't make sense to use theme driven colors until/unless we use dark-mode map tiles, which provide the relevant background in this case.

Yes- my thoughts exactly re: greyscale - in #4298 , we have already expanded our greyscale significantly. However - I do think its worth entertaining a dark mode map tile - since those lighter colors can be very very bright in a dark setting. Happy to work with maps team to investigate.

**Review by joshuarrrr — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Pull request merged**
